report/renderer/i18n.js

```diff
@@ -32,36 +32,42 @@ export class I18n {
 
   /**
    * @param {number} number
-   * @param {number} granularity
+   * @param {number|undefined} granularity
    * @param {Intl.NumberFormatOptions} opts
    * @return {string}
    */
   _formatNumberWithGranularity(number, granularity, opts = {}) {
-    opts = {...opts};
-    const log10 = -Math.log10(granularity);
-    if (!Number.isFinite(log10) || (granularity > 1 && Math.floor(log10) !== log10)) {
-      throw new Error(`granularity of ${granularity} is invalid`);
-    }
+    if (granularity !== undefined) {
+      opts = {...opts};
+      const log10 = -Math.log10(granularity);
+      if (!Number.isFinite(log10) || (granularity > 1 && Math.floor(log10) !== log10)) {
+        throw new Error(`granularity of ${granularity} is invalid`);
+      }
 
-    if (granularity < 1) {
-      opts.minimumFractionDigits = opts.maximumFractionDigits = Math.ceil(log10);
-    }
+      if (granularity < 1) {
+        opts.minimumFractionDigits = opts.maximumFractionDigits = Math.ceil(log10);
+      }
 
-    number = Math.round(number / granularity) * granularity;
+      number = Math.round(number / granularity) * granularity;
 
-    // Avoid displaying a negative value that rounds to zero as "0".
-    if (Object.is(number, -0)) number = 0;
+      // Avoid displaying a negative value that rounds to zero as "0".
+      if (Object.is(number, -0)) number = 0;
+    } else if (Math.abs(number) <= 0.0001) {
```

Comment:
Can you add a comment explaining where 0.0001 comes from.

Reply:
I think it also needs to be `Math.abs(number) < 0.0005` (less than half the smallest expressible value).

e.g. with this `_formatNumberWithGranularity(-0.0004)` currently gives `'-0'`